test/CMakeLists.txt

```diff
@@ -248,6 +248,10 @@ if (TILEDB_VERBOSE)
   add_definitions(-DTILEDB_VERBOSE)
 endif()
 
+if (TILEDB_EXPERIMENTAL_FEATURES)
+  add_definitions(-DTILEDB_EXPERIMENTAL_FEATURES)
+endif()
```

Comment:
This is going to need to be added to all object files as well. As discussed in the debug session this morning, it might be worth taking this opportunity to switch to `target_compile_definitions` (see https://cmake.org/cmake/help/latest/command/target_compile_definitions.html) and creating an interface library for compiler definitions and options.